connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogTableSpec.scala

```diff
@@ -0,0 +1,343 @@
+
+    spark.sql(
+      s"""ALTER TABLE $defaultKs.$testTable SET TBLPROPERTIES ('gc_grace_seconds' = '22')"""
+    )
+    val result = getTable(defaultKs, testTable)
+      .getOptions
+      .get(fromInternal("gc_grace_seconds"))
+      .asInstanceOf[Int].toString
+
+    result should be("22")
+  }
+
+  it should "alter map table properties" in {
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+
+    spark.sql(
+      s"""ALTER TABLE $defaultKs.$testTable SET TBLPROPERTIES (
+         |'compaction'='{class=SizeTieredCompactionStrategy,bucket_high=1001}'
+         |)""".stripMargin
+    )
+
+    val compactionOptions = getTable(defaultKs, testTable)
+      .getOptions.asScala
+      .get(fromInternal("compaction"))
+      .get
+      .asInstanceOf[java.util.Map[String, AnyRef]]
+      .asScala
+
+    compactionOptions.get("bucket_high").get.toString should be ("1001")
+  }
+
+  it should "alter add columns" in {
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+
+    spark.sql(
+      s"""ALTER TABLE $defaultKs.$testTable ADD COLUMNS (newCol INT) """.stripMargin
+    )
+
+    getTable(defaultKs, testTable)
+      .getColumn(fromInternal("newCol"))
+      .get().getName.asInternal() shouldBe ("newCol")
+  }
+
+  it should "alter remove columns" in {
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+
+    spark.sql(
+      s"""ALTER TABLE $defaultKs.$testTable DROP COLUMNS (value) """.stripMargin
+    )
+
+    getTable(defaultKs, testTable)
+      .getColumn(fromInternal("value"))
+      .isPresent shouldBe (false)
+  }
+
+  it should "throw exceptions when removing illegal columns" in {
+    createDefaultKs()
+    spark.sql(
+      s"""CREATE TABLE $defaultKs.$testTable (
+         |key_1 Int, key_2 Int, key_3 Int,
+         |cc1 STRING, cc2 String, cc3 String,
+         |value String) USING cassandra
+         |PARTITIONED BY (key_1, key_2, key_3)
+         |TBLPROPERTIES (clustering_key='cc1.asc, cc2.desc, cc3.asc')""".stripMargin)
+
+    val excep = intercept[SparkException] {
```

Comment:
In other cases we have a more precise type to indicate specific error cases.  Worth something like a NoSuchTableException for this case?  SparkException seems... awfully general here.

Reply:
Yeah I have no Idea why this one is like that

Reply:
Ok having run this again I know why this is like this. The exception thrown is 
SparkException( CassandraCatalogException ) so this is correct. Spark wraps our catalog exception in it's own before surfacing it.